modules/ROOT/pages/sap/sap-s4-hana-connector-reference.adoc

```diff
@@ -0,0 +1,518 @@
+|======================
+
+==== For Configurations.
+* <<config>> {nbsp}
+
+==== Throws
+* S4HANA:TIMEOUT {nbsp}
+* S4HANA:CONNECTIVITY {nbsp}
+* S4HANA:NO_SUCH_ENTITY_FIELD {nbsp}
+* S4HANA:NO_SUCH_SERVICE {nbsp}
+* S4HANA:NO_SUCH_ENTITY_TYPE {nbsp}
+* S4HANA:INVALID_KEY {nbsp}
+* S4HANA:NO_SUCH_ENTITY_KEY {nbsp}
+* S4HANA:RETRY_EXHAUSTED {nbsp}
+* S4HANA:UNAUTHORIZED {nbsp}
+* S4HANA:INVALID_ENTITY {nbsp}
+
+
+[[query]]
+=== Query
+`<s4hana:query>`
+
++++
+Executes a retrieve operation on the SAP S/4HANA instance filtering the results by the parameters defined. If no filter is added, then all results will be listed. This is a paginated operation.
++++
+
+==== Parameters
+[cols=".^20%,.^20%,.^35%,.^20%,^.^5%", options="header"]
+|======================
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | *x*{nbsp}
+| Streaming Strategy a| * <<repeatable-in-memory-iterable>>
+* <<repeatable-file-store-iterable>>
+* <<non-repeatable-iterable>> |  +++Configure if repeatable streams should be used and their behaviour+++ |  | {nbsp}
+| Service a| String |  +++The type of service. This is defined from the services available in the SAP S/4HANA instance.+++ |  | *x*{nbsp}
+| Entity type a| String |  +++The type of entity over which the operation is executed. This is defined from the types of entities available in the selected service.+++ |  | *x*{nbsp}
+| Returned Fields a| String |  +++The comma separated list of fields of the entity to return. If no field is selected, then all fields are returned.+++ |  +++*+++ | {nbsp}
+| Filter a| String |  +++The condition to filter the resulting list of entities.+++ |  | {nbsp}
+| Order by a| String |  +++Allows to request information in either ascending or descending order by using the ?asc? or ?desc? suffixes. If asc or desc not specified, then the resources will be ordered in ascending order.+++ |  | {nbsp}
+| Page size a| Number |  +++The size of the pages retrieved by the query.+++ |  +++100+++ | {nbsp}
+| Maximum returned elements a| Number |  +++Limits the amount of records to be retrieved by this query.+++ |  | {nbsp}
+| Skipped record amount a| Number |  +++The amount of records to skip before starting returning results.+++ |  +++0+++ | {nbsp}
+| Custom headers a| Object |  +++Custom headers that the request will include. The ones specified here will overwrite the default ones specified in the config.+++ |  | {nbsp}
+| Custom query parameters a| Object |  +++Custom query parameters that the request will include. The ones specified here will overwrite the default ones specified in the config.+++ |  | {nbsp}
+| Target Variable a| String |  +++The name of a variable on which the operation's output will be placed+++ |  | {nbsp}
+| Target Value a| String |  +++An expression that will be evaluated against the operation's output and the outcome of that expression will be stored in the target variable+++ |  +++#[payload]+++ | {nbsp}
```

Comment:
An expression that will be evaluated against the operation's output. The outcome of that expression will be stored in the target variable.